src/distilabel/llm/vertexai.py

```diff
@@ -0,0 +1,173 @@
+            )
+
+    @property
+    def model_name(self) -> str:
+        """Returns the model name used for the LLM."""
+        return self.model
+
+    def _chattype_to_content(self, input: "ChatType") -> List["Content"]:
+        """Converts a chat type to a list of content items expected by the API.
+
+        Args:
+            input: the chat type to be converted.
+
+        Returns:
+            List[str]: a list of content items expected by the API.
+        """
+        contents = []
+        for message in input:
+            if message["role"] not in ["user", "model"]:
+                raise ValueError(
+                    "`VertexAILLM only supports the roles 'user' or 'model'."
+                )
+            contents.append(
+                self._content_class(
+                    role=message["role"],
+                    parts=[self._part_class.from_text(message["content"])],
+                )
+            )
+        return contents
+
+    async def agenerate(  # type: ignore
+        self,
+        input: "ChatType",
+        num_generations: int = 1,
+        temperature: Optional[float] = None,
+        top_p: Optional[float] = None,
+        top_k: Optional[int] = None,
+        max_output_tokens: Optional[int] = None,
+        stop_sequences: Optional[List[str]] = None,
+        safety_settings: Optional["SafetySettingsType"] = None,
+        tools: Optional[List["Tool"]] = None,
+    ) -> "GenerateOutput":
+        """Generates `num_generations` responses for the given input using the [VertexAI async client definition](https://cloud.google.com/vertex-ai/docs/generative-ai/model-reference/gemini).
+
+        Args:
+            input: a single input in chat format to generate responses for.
+            num_generations: the number of generations to create per input. Defaults to
+                `1`.
+            temperature: Controls the randomness of predictions. Range: [0.0, 1.0]. Defaults to `None`.
+            top_p: If specified, nucleus sampling will be used. Range: (0.0, 1.0]. Defaults to `None`.
+            top_k: If specified, top-k sampling will be used. Defaults to `None`.
+            max_output_tokens: The maximum number of output tokens to generate per message. Defaults to `None`.
+            stop_sequences: A list of stop sequences. Defaults to `None`.
+            safety_settings: Safety configuration for returned content from the API. Defaults to `None`.
+            tools: A potential list of tools that can be used by the API. Defaults to `None`.
+
+        Returns:
+            A list of lists of strings containing the generated responses for each input.
+        """
+        contents = self._chattype_to_content(input)
+        content = await self._aclient.generate_content_async(
+            contents=contents,
+            generation_config=self._generation_config_class(
+                candidate_count=num_generations,
```

Comment:
Do you know if they have updated the `candidate_count` parameter and now more than 1 generation is allowed?

Reply:
nope, I assumed it was.